torch/distributed/rpc/__init__.py

```diff
@@ -6,15 +6,19 @@
 
 
 if sys.version_info >= (3, 0):
+    from . import api
     from .api import _init_rpc
     from .api import *  # noqa: F401
     import torch.distributed.autograd
 
-    def init_model_parallel(self_name,
-                            backend=RpcBackend.PROCESS_GROUP,
-                            self_rank=-1,
-                            init_method=None,
-                            num_send_recv_threads=4):
+    def init_model_parallel(
+        self_name,
+        backend=RpcBackend.PROCESS_GROUP,
+        init_method=None,
+        self_rank=-1,
+        worker_name_to_id=None,
```

Comment:
It seems the `Store.hpp` API only allows `int64_t` value type. @pietern will it be acceptable if we modify the store API to take string values?

Reply:
We can't pass `world_size` only, because for c10d::Store, there is no "list" API to list keys of other participant nodes. There is no way to wait until the number participants equals `world_size`.

Reply:
Or we use the "add" API as an atomic counter? Every one publish it's only address and add the counter?

Reply:
If this PR is necessary to unblock, let's get this in, but would prefer to not having a `worker_name_to_id` arg in the final API. We can discuss how to modify stores to handle this use case.

Reply:
I agree Shen this API is getting cluttered, we need to clean this up a bit.

Reply:
@satgera 

Made a first step for cleaning this up in #28628.